File: sample-code/fibonacci.cpp
// Fibonacci sequence generator (iterative)
#include <iostream>
#include <vector>

std::vector<long long> fib(int n) {
    std::vector<long long> out;
    out.reserve(n);
    long long a = 0;
    long long b = 1;
    for (int i = 0; i < n; i++) {
        out.push_back(a);
        long long next = a + b;
        a = b;
        b = next;
    }
    return out;
}

int main() {
    int n = 12;
    auto out = fib(n);
    for (size_t i = 0; i < out.size(); i++) {
        std::cout << out[i] << (i + 1 == out.size() ? "\n" : ", ");
    }
    return 0;
}

File: sample-code/fibonacci.rs
// Fibonacci sequence generator (iterative)

fn fib(n: usize) -> Vec<u64> {
    let mut out: Vec<u64> = Vec::with_capacity(n);
    let mut a: u64 = 0;
    let mut b: u64 = 1;
    for _ in 0..n {
        out.push(a);
        let next = a + b;
        a = b;
        b = next;
    }
    out
}

fn main() {
    let n = 12;
    let out = fib(n);
    let s = out
        .iter()
        .map(|v| v.to_string())
        .collect::<Vec<_>>()
        .join(", ");
    println!("{}", s);
}
